tests/bluetooth/tester/src/btp_aics.c

```diff
@@ -0,0 +1,608 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stddef.h>
+#include <errno.h>
+
+#include <zephyr/types.h>
+#include <zephyr/kernel.h>
+
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/audio/audio.h>
+#include <zephyr/bluetooth/audio/micp.h>
+#include <zephyr/bluetooth/audio/aics.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/byteorder.h>
+
+#include "bap_endpoint.h"
+#include "btp/btp.h"
+
+#define LOG_MODULE_NAME bttester_aics
+LOG_MODULE_REGISTER(LOG_MODULE_NAME, CONFIG_BTTESTER_LOG_LEVEL);
+
+#define BT_AICS_MAX_INPUT_DESCRIPTION_SIZE 16
+#define BT_AICS_MAX_OUTPUT_DESCRIPTION_SIZE 16
+
+struct btp_aics_instance aics_client_instance;
+struct btp_aics_instance aics_server_instance;
+
+static struct net_buf_simple *rx_ev_buf = NET_BUF_SIMPLE(BT_AICS_MAX_INPUT_DESCRIPTION_SIZE +
+							 sizeof(struct btp_aics_description_ev));
+
+static uint8_t aics_supported_commands(const void *cmd, uint16_t cmd_len,
+				       void *rsp, uint16_t *rsp_len)
+{
+	struct btp_aics_read_supported_commands_rp *rp = rsp;
+
+	/* octet 0 */
+	tester_set_bit(rp->data, BTP_AICS_READ_SUPPORTED_COMMANDS);
+	tester_set_bit(rp->data, BTP_AICS_SET_GAIN);
+	tester_set_bit(rp->data, BTP_AICS_MUTE);
+	tester_set_bit(rp->data, BTP_AICS_UNMUTE);
+	tester_set_bit(rp->data, BTP_AICS_MAN_GAIN);
+	tester_set_bit(rp->data, BTP_AICS_AUTO_GAIN);
+	tester_set_bit(rp->data, BTP_AICS_MAN_GAIN_ONLY);
+
+	/* octet 1 */
+	tester_set_bit(rp->data, BTP_AICS_AUTO_GAIN_ONLY);
+	tester_set_bit(rp->data, BTP_AICS_AUDIO_DESCRIPTION_SET);
+	tester_set_bit(rp->data, BTP_AICS_MUTE_DISABLE);
+	tester_set_bit(rp->data, BTP_AICS_GAIN_SETTING_PROP);
+	tester_set_bit(rp->data, BTP_AICS_TYPE);
+	tester_set_bit(rp->data, BTP_AICS_STATUS);
+	tester_set_bit(rp->data, BTP_AICS_STATE);
+
+	/* octet 2 */
+	tester_set_bit(rp->data, BTP_AICS_DESCRIPTION);
+
+	*rsp_len = sizeof(*rp) + 2;
+
+	return BTP_STATUS_SUCCESS;
+}
+
+void btp_send_aics_state_ev(struct bt_conn *conn, int8_t gain, uint8_t mute, uint8_t mode)
+{
+	struct btp_aics_state_ev ev;
+	struct bt_conn_info info;
+
+	(void)bt_conn_get_info(conn, &info);
+	bt_addr_le_copy(&ev.address, info.le.dst);
+
+	ev.gain = gain;
+	ev.mute = mute;
+	ev.mode = mode;
+
+	tester_event(BTP_SERVICE_ID_AICS, BTP_AICS_STATE_EV, &ev, sizeof(ev));
+}
+
+void btp_send_gain_setting_properties_ev(struct bt_conn *conn, uint8_t units, int8_t minimum,
+					 int8_t maximum)
+{
+	struct btp_gain_setting_properties_ev ev;
+	struct bt_conn_info info;
+
+	(void)bt_conn_get_info(conn, &info);
+	bt_addr_le_copy(&ev.address, info.le.dst);
+
+	ev.units = units;
+	ev.minimum = minimum;
+	ev.maximum = maximum;
+
+	tester_event(BTP_SERVICE_ID_AICS, BTP_GAIN_SETTING_PROPERTIES_EV, &ev, sizeof(ev));
+}
+
+void btp_send_aics_input_type_event(struct bt_conn *conn, uint8_t input_type)
+{
+	struct btp_aics_input_type_ev ev;
+	struct bt_conn_info info;
+
+	(void)bt_conn_get_info(conn, &info);
+	bt_addr_le_copy(&ev.address, info.le.dst);
```

Comment:
If you just need the `dst` address, then I suggest



```suggestion

	bt_addr_le_copy(&ev.address, bt_conn_get_dst(conn));
```

Applies to multiple places